kvserver/rangefeed: metamorphically enable RangefeedUseBufferedSender

wenyihu6 opened the pull request:
**kvserver/rangefeed: rename .buffered_stream_sender. to .buffered_sender.**

This patch renames kv.rangefeed.buffered_stream_sender.enabled to
kv.rangefeed.buffered_sender.enabled to align with variable and
struct names better.

Epic: none
Release: none

---

**kvserver/rangefeed: metamorphically enable RangefeedUseBufferedSender**

This patch metamorphically enables the cluster setting
RangefeedUseBufferedSender.

Release note: none
Part of: https://github.com/cockroachdb/cockroach/issues/135332

wenyihu6 committed:
kvserver/rangefeed: metamorphically enable RangefeedUseBufferedSender

This patch metamorphically enables the cluster setting
RangefeedUseBufferedSender.

Release note: none
Part of: #135332

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/136995)
<!-- Reviewable:end -->

wenyihu6 commented:
How do we feel about lifting the restriction here https://github.com/cockroachdb/cockroach/blob/a9076591f5bd9841040fea644bca63465103fa3a/pkg/kv/kvserver/replica_rangefeed.go#L102-L107?

wenyihu6 committed:
kvserver/rangefeed: rename *.buffered_stream_sender.* to *.buffered_sender.*

This patch renames kv.rangefeed.buffered_stream_sender.enabled to
kv.rangefeed.buffered_sender.enabled to align with variable and
struct names better.

Epic: none
Release: none

stevendanna commented on pkg/kv/kvserver/replica_rangefeed.go:
```diff
@@ -95,10 +96,10 @@ var ErrBufferedSenderNotSupported = unimplemented.NewWithIssue(
 // unimplemented and disabled everywhere (#126560).
```
Let's update the comment while we are here. This is no longer unimplemented :D.

wenyihu6 replied:
🥳 Good catch, done.

stevendanna reviewed (approved):
Thanks!

wenyihu6 commented:
TFTR!

bors r=stevendanna

wenyihu6 commented:
Extended CI failure TestLogGrowthWhenRefreshingPendingCommands looks unrelated.
